src/validations/vulnerable-dependencies.js

```diff
@@ -1,32 +1,111 @@
     defaultParam: true,
 
-    configurator (currentVal) {
+    configurator(currentVal) {
         return confirm(
             "Would you like to verify that your package doesn't have vulnerable dependencies before publishing?",
             currentVal
         );
     },
 
-    run () {
+    run() {
         return new Promise((resolve, reject) => {
-            const pkg = pathJoin(process.cwd(), 'package.json');
+            const projectDir = pathJoin(process.cwd());
+            const defaultArgs = nsp.sanitizeParameters({});
+            const args = {
+                baseUrl: defaultArgs.baseUrl,
+                proxy: defaultArgs.proxy,
+                reporter: 'summary',
+                'warn-only': false,
+                path: projectDir,
+                pkg: readPkg.sync(projectDir, { normalize: false }),
+                offline: false,
+                exceptions: [],
+            };
 
-            nsp.check({ package: pkg }, (err, data) => {
-                if (err || data.length > 0)
-                    reject(nsp.formatters.summary(err, data));
-                else
+            nsp
+                .check(args)
+                .then((result) => {
+                    if (vulnerabilitiesFoundIn(result)) {
+                        const errs = result.data
+                            .map((vulnerability) => summaryOf(vulnerability))
+                            .sort();
+                        reject(errs);
+                        return;
+                    }
                     resolve();
-            });
+                })
+                .catch((err) => {
+                    reject(err);
+                });
         });
-    }
+    },
 };
+
+function vulnerabilitiesFoundIn(result) {
+    return result && result.data && result.data.length > 0;
+}
+
+function summaryOf(vulnerability) {
+    const vulnerablePackageName = `${vulnerability.module ||
+        'undefined'}@${vulnerability.version || '?.?.?'}`;
+    const vulnerablePackagePath =
+        vulnerability.path && vulnerability.path.length >= 2
+            ? vulnerability.path.slice(1)
+            : [];
+
+    const rootPackageName =
+        vulnerablePackagePath.length >= 1
+            ? vulnerablePackagePath[0]
+            : vulnerablePackageName;
+
+    const recommendation = vulnerability.recommendation
+        ? vulnerability.recommendation.replace('\n', '\n\t')
+        : '';
+
+    const vulnerabilityIsDirectDependency =
+        vulnerablePackageName === rootPackageName;
+    // prettier-ignore
+    const summary = vulnerabilityIsDirectDependency
+        ? `Vulnerability found in ${elegant(rootPackageName)}\n\t${recommendation}\n\tAdvisory: ${vulnerability.advisory || ''}`
+        : `Vulnerability found in ${chalk.bold(rootPackageName)}\n\tinside ${elegant(vulnerablePackagePath)}\n\t${vulnerability.recommendation || ''}\n\tAdvisory: ${vulnerability.advisory || ''}`;
+    return summary;
+}
+
+function elegant(pathOrName) {
```

Comment:
This looks like a good candidate to be moved to some separate utils file or even separate package. Don't bother doing it in context of this PR, it's just a suggestion.

Reply:
@inikulin ,fully agree with you. I put it in my todo list